cmd/reconciler/main.go

```diff
@@ -98,8 +98,8 @@ var (
 
 	renderingEnabled  = flag.Bool("rendering-enabled", util.EnvBool(reconcilermanager.RenderingEnabled, false), "")
 	namespaceStrategy = flag.String(flags.namespaceStrategy, util.EnvString(reconcilermanager.NamespaceStrategy, ""),
```

Comment:
I was intending to leave it as setting the default in `main` for consistency with the existing code. Would you prefer a refactor?

Reply:
After taking a closer look, I think it may have been written this way to support different default behavior for RootSync/RepoSync reconcilers.

For RootSync reconcilers, we set the default value if unset.

For RepoSync reconcilers, we log an error if the value is set.

If we set the default in the flag above, I don't think we would be able to distinguish between these two behaviors in the logic below. With that in mind, I suggest we leave the logic as-is for now